### PR TITLE
wait for mdnsd to start

### PR DESCRIPTION
**Pull request by jhixson74**
Ticket: #26531

### CODE REVIEW

File: src/middlewared/middlewared/plugins/mdns.py [hunk 1]
```diff
@@ -14,6 +16,76 @@
...
 
+class mDNSDaemonThread(threading.Thread):
+    def __init__(self, *args, **kwargs):
+        super(mDNSDaemonThread, self).__init__()
+        self.setDaemon(True)
+        self.mdnsd_pidfile = "/var/run/mdnsd.pid"
+        self.mdnsd_piddir = "/var/run/"
+
+    def pidfile_exists(self):
+        return os.access(self.mdnsd_pidfile, os.F_OK)
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/mdns.py]: Why not `os.path.exists`?

File: src/middlewared/middlewared/plugins/mdns.py [hunk 2]
```diff
@@ -14,6 +16,76 @@
...
+            procname = str(procdict['name'])
+
+            match = re.search("([^/]+)$", procname)
+            if not match:
+                continue
+
+            procname = match.group(0)
+            if procname == "mdnsd":
+                mdnsd_proc = proc
+                break
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/mdns.py]: It's hard to understand what you're trying to achieve here. Is it `procname.split("/")[0] == "mdnsd"`?

File: src/middlewared/middlewared/plugins/mdns.py [hunk 3]
```diff
@@ -15,6 +17,74 @@
...
+
+        if mdnsd_proc and mdnsd_proc.is_running():
+            alive = True
+
+        return alive
+
+    def wait(self):
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/mdns.py]: `wait` for what? Maybe it should be `wait_startup`?

File: src/middlewared/middlewared/plugins/mdns.py [hunk 4]
```diff
@@ -81,6 +151,9 @@ def __init__(self, **kwargs):
         self.logger = kwargs.get('logger')
         self.timeout = kwargs.get('timeout', 30)
 
+        self.mdnsd = mDNSDaemonThread()
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/mdns.py]: Why is it called `mDNSDaemonThread`? Name like `mDNSStartupWaiter` will have more sense.

**Comment by kmoore134** [comment]
John, do you want to make any adjustments based on the feedback here?

File: src/middlewared/middlewared/plugins/mdns.py (hunk 1 above)
Comment (jhixson74) [reply src/middlewared/middlewared/plugins/mdns.py]: Why not os.access ?

File: src/middlewared/middlewared/plugins/mdns.py (hunk 2 above)
Comment (jhixson74) [reply src/middlewared/middlewared/plugins/mdns.py]: What is difficult to understand? It's a very simple regular expression.

File: src/middlewared/middlewared/plugins/mdns.py (hunk 3 above)
Comment (jhixson74) [reply src/middlewared/middlewared/plugins/mdns.py]: Wait for mDNSResponder to startup

File: src/middlewared/middlewared/plugins/mdns.py (hunk 4 above)
Comment (jhixson74) [reply src/middlewared/middlewared/plugins/mdns.py]: Because it was formerly a thread. Code was rearranged a little bit.

**Pull request merged**